File: exercise/python/day24/domain/toy_created_event.py
import datetime
import uuid

from domain.core.event import Event
from domain.stock_unit import StockUnit


class ToyCreatedEvent(Event):
    def __init__(self, event_id: uuid.UUID, date: datetime, name: str, stock: StockUnit):
        super().__init__(event_id, date)
        self.name = name
        self.stock = stock

File: exercise/ts/day24/src/domain/ToyCreatedEvent.ts
import {Event} from "./core/Event"
import {StockUnit} from "./StockUnit";

export class ToyCreatedEvent extends Event {
    constructor(
        id: string,
        date: Date,
        public name: string,
        public stock: StockUnit
    ) {
        super(id, 1, date);
    }
}
